### DISCUSSION TITLE
Annotations for iOS Quick Look

### DISCUSSION THREAD

**Discussion by itssmutnuri** [google/model-viewer#2151] [Q&A]
Hi,

I was trying to add some annotations to the Astronaut model (referring the [docs](https://modelviewer.dev/examples/annotations/index.html)) and was not able to figure out how to pull the same for Quick Look. 

```html
<model-viewer id="modelContainer"
        src="/assets/Astronaut.glb"
        alt="A 3D model of an astronaut" shadow-intensity="1" 
        ar ar-scale="auto" camera-controls
        style="--poster-color:#ffffff00;" alt="A 3D model of an astronaut."
        ios-src="/assets/Astronaut.usdz">
        <!-- button style sheets have been added from css-->
        <button slot="hotspot-hand" data-position="-0.54 0.93 0.1" data-normal="-0.73 0.05 0.69">
            <div id="annotation">This is an example of a hotspot</div>
        </button>
</model-viewer>
```

While the annotation cards appear in the 3D view, Quick Look doesn't show them - just the model comes up. Can anyone please help with this? Is it currently possible?

Thanks!

**Comment by elalish** [comment]
Sorry, this is not possible as QuickLook is a native app and does not render arbitrary HTML like the browser does. This is why we hope WebXR will come to Safari, so everything can be consistent.

**Reply by elalish** [reply]
@grorg This is the kind of complex 2D-DOM-3D-Model API that we have built, which is easy with existing web tech, but would be a real bear to standardize, because we have made a host of arbitrary UX choices here.

**Comment by rebrob** [comment]
Hello,

Have there been any updates to QuickLook that enable annotations in model-viewer on ios?

**Reply by elalish** [reply]
Nope; actually I haven't seen any updates to Quick Look in years now, aside from occasional breaking changes to USDZ.
